src/Microsoft.Azure.SignalR.AspNet/Infrastructure/ServiceConnection.cs

```diff
@@ -132,17 +128,47 @@ private Task OnConnectedAsyncCore(OpenConnectionMessage message)
             {
                 var transport = _clientConnectionManager.CreateConnection(message, this);
                 _clientConnections.TryAdd(transport.ConnectionId, transport);
+
+                _ = ProcessOutgoingMessagesAsync(connectionId);
                 Log.ConnectedStarting(_logger, connectionId);
                 return Task.CompletedTask;
             }
             catch (Exception e)
             {
                 Log.ConnectedStartingFailed(_logger, connectionId, e);
-                PerformDisconnectCore(connectionId);
+                _ = PerformDisconnectAsyncCore(connectionId);
                 return WriteAsync(new CloseConnectionMessage(connectionId, e.Message));
             }
         }
 
+        private async Task ProcessOutgoingMessagesAsync(string connectionId)
+        {
+            if (_clientConnections.TryGetValue(connectionId, out var transport))
+            {
+                try
+                {
+                    // Check if channel is closed
+                    while (await transport.Input.WaitToReadAsync())
```

Comment:
I recommend following the pattern we use in AspNetCore
https://github.com/aspnet/SignalR/blob/2d4fb0af6fd2ef2e3a059a2a15a56484d8800d35/src/Microsoft.AspNetCore.SignalR.Client.Core/HubConnectionExtensions.StreamAsChannelAsync.cs#L275-L277